ibm_cloud_sdk_core/utils.py

```diff
@@ -125,15 +134,17 @@ def read_from_credential_file(service_name, separator='='):
     return config
 
 def _parse_key_and_update_config(config, service_name, key, value):
-    if service_name in key:
-        index = key.find('_')
+    service_name = service_name.replace(' ', '_').replace('-', '_').lower()
+    key = key.replace(' ', '_').replace('-', '_').lower()
+    if key.startswith(service_name):
+        index = key.find(service_name)
         if index != -1:
-            config[key[index + 1:]] = value
+            config[key[index + len(service_name) + 1:]] = value
 
 def read_from_vcap_services(service_name):
-    service_name = service_name.replace(' ', '_').lower()
+    service_name = service_name.replace(' ', '_').replace('-', '_').lower()
```

Comment:
When we get config info from VCAP_SERVICES, we should not do any transformations on `service_name` at all.
This is because the exact service name is configured in the API definition and is inserted into the generated code and should be used as-is.   So, a vcap services name obtained from the API definition of `personality-insights` is the exact string that we'd expect to find within VCAP_SERVICES.

Reply:
I perform the key conversion here: https://github.com/IBM/python-sdk-core/pull/28/files#diff-7048c45d573bb78c4bac9b0291b4e065R152 so that when service_name = `personality-insights` and vcap name is `personality-insights` would both be converted to `personality_insights`

Reply:
Why?   All you need to do is compare the "as-is" service_name as it was generated in the service and compare it with the outer-most map key within the VCAP_SERVICES json object.   By trying to do all these extra, unnecessary transformations, you're actually accepting a different "language" than the other cores.